### PR TITLE
Adding helper for starting thread in teams

### PR DESCRIPTION
**Pull request by Virtual-Josh**

### CODE REVIEW

File: libraries/botbuilder/tests/teamsInfo.test.js [hunk 1]
```diff
@@ -26,6 +26,43 @@ class TestContext extends TurnContext {
 }
 
 describe('TeamsInfo', () => {
+    describe('sendMessageToTeamsChannel()', () =>{
```
Comment (johnataylor) [inline libraries/botbuilder/tests/teamsInfo.test.js]: There are negative tests for the args. Can we add a positive test for the appropriate http call actually being made?
You will have to use nock to set up the expected response to the particular http call that is being made.

File: libraries/botbuilder/src/teamsInfo.ts [hunk 2]
```diff
@@ -31,6 +35,36 @@ export class TeamsInfo {
         return await this.getTeamsConnectorClient(context).teams.fetchTeamDetails(t);
     }
 
+    public static async sendMessageToTeamsChannel(context: TurnContext, activity: Partial<Activity>, teamsChannelId: string): Promise<[Partial<ConversationReference>, string]> {
+        if (context == null){
```
Comment (johnataylor) [inline libraries/botbuilder/src/teamsInfo.ts]: This does not follow our typical JavaScript conventions. You should test for Falsey. i.e. you can use if (!context) etc.

File: libraries/botbuilder/src/teamsInfo.ts [hunk 3]
```diff
@@ -31,6 +35,36 @@ export class TeamsInfo {
...
+    public static async sendMessageToTeamsChannel(context: TurnContext, activity: Partial<Activity>, teamsChannelId: string): Promise<[Partial<ConversationReference>, string]> {
+        if (context == null){
+            throw new Error("TurnContext cannot be null");
+        }
+
+        if (activity == null){
+            throw new Error("Activity cannot be null");
+        }
+
+        if (teamsChannelId == null || teamsChannelId == ""){
```
Comment (johnataylor) [inline libraries/botbuilder/src/teamsInfo.ts]: in particular this test is simplified with an understanding of Falsey.

File: libraries/botbuilder/src/teamsInfo.ts (hunk 2 above)
Comment (johnataylor) [reply libraries/botbuilder/src/teamsInfo.ts]: ...and all these if statements are missing a space between the closing parenthesis and the opening curly brace.

**Comment by johnataylor** [comment]
fixes #1631

File: libraries/botbuilder/tests/teamsInfo.test.js (hunk 1 above)
Comment (Virtual-Josh) [reply libraries/botbuilder/tests/teamsInfo.test.js]: added.

File: libraries/botbuilder/src/teamsInfo.ts [hunk 4]
```diff
@@ -31,6 +35,36 @@ export class TeamsInfo {
         return await this.getTeamsConnectorClient(context).teams.fetchTeamDetails(t);
     }
 
+    public static async sendMessageToTeamsChannel(context: TurnContext, activity: Partial<Activity>, teamsChannelId: string): Promise<[Partial<ConversationReference>, string]> {
+        if (!context) {
```
[the remark this answers is not in the document]
Comment (Virtual-Josh) [reply libraries/botbuilder/src/teamsInfo.ts]: updated.

File: libraries/botbuilder/src/teamsInfo.ts [hunk 5]
```diff
@@ -31,6 +35,36 @@ export class TeamsInfo {
...
+
+        const convoParams = <ConversationParameters>{
+            isGroup: true,
+            channelData: {
+                channel: {
+                    id: teamsChannelId
+                }
+            },
+            activity: activity
+        }
```
[the remark this answers is not in the document]
Comment (Virtual-Josh) [reply libraries/botbuilder/src/teamsInfo.ts]: updated

**Pull request merged**